src/java/org/apache/cassandra/db/partitions/UnfilteredPartitionIterators.java

```diff
@@ -57,10 +57,10 @@ public static UnfilteredRowIterator getOnlyElement(final UnfilteredPartitionIter
         // If the query has no results, we'll get an empty iterator, but we still
         // want a RowIterator out of this method, so we return an empty one.
         UnfilteredRowIterator toReturn = iter.hasNext()
-                              ? iter.next()
-                              : EmptyIterators.unfilteredRow(command.metadata(),
-                                                             command.partitionKey(),
-                                                             command.clusteringIndexFilter().isReversed());
+                                         ? iter.next()
```

Comment:
@justinchuch Can you review this commit please. It contains a number of small changes needed for SAI integration related to partition / row iteration.